### PR TITLE
Warn that `Update-Package -Reinstall` is not relevant for PackageReference based projects

### LINKED ISSUE #6088: Update-Package -reinstall -ProjectName <project> does not work for PR
```
PM> get-package -projectname ClassLibrary1

Id                                  Versions                                 ProjectName                                                                                                                                                                                                                                               
--                                  --------                                 -----------                                                                                                                                                                                                                                               
newtonsoft.json                     {10.0.3}                                 ClassLibrary1                                                                                                                                                                                                                                             
NETStandard.Library                 {2.0.0}                                  ClassLibrary1                                                                                                                                                                                                                                             


PM> Update-Package -reinstall -projectname ClassLibrary1
No package updates are available from the current package source for project 'ClassLibrary1'.
Executing nuget actions took 0 ms
Time Elapsed: 00:00:00.2839360
```

### PR DESCRIPTION
**Pull request by nkolev92**
## Bug

Fixes: https://github.com/NuGet/Home/issues/6088

Regression? Last working version:

## Description

Refer to the linked issue: https://github.com/NuGet/Home/issues/6088 and https://docs.microsoft.com/en-us/nuget/consume-packages/reinstalling-and-updating-packages

Update-Package -Reinstall is a way to fix up broken refs for packages.config based projects. 

PackageReference is build integrated and as such, restore and build cooperate to ensure all the refs are correctly wired up. 
Furthermore PackageReference does not affect the project structure itself, but only the output. 

* Whenever the reinstall option is invoked on a PackageReference (or project.json) based project, warn that the operation is *not supported*. 
* Note that the message says `PackageReference` explicitly, but there is a slight chance the project is project.json. This is not an extremely likely scenario, so I don't see harm from calling out `PackageReference explicitly. 

cc @chgill-MSFT @aortiz-msft, I'd appreciate any feedback you may have.

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [x] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [x] Documentation PR or issue filled - https://github.com/NuGet/docs.microsoft.com-nuget/issues/2519
  - **OR**
  - [ ] N/A

### CODE REVIEW

File: src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs
Comment (erdembayar) [inline src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs]: Previously I received feedback about below is better for performance, maybe you can check.

https://github.com/NuGet/NuGet.Client/blob/77e1bdb266093fd1eafff90ef21ad55a2a448889/src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs#L2055-L2062

File: src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs
Comment (erdembayar) [inline src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs]: This one looks exactly same as above could we make separate method and just call from both places.

Comment (nkolev92) [reply src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs]: It's very similar, I'll refactor once the concept has been approved.

**Comment by nkolev92** [comment]
cc @NuGet/nuget-client and @chgill-MSFT @aortiz-msft, can you please take a look?

File: src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs
Comment (aortiz-msft) [inline src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs]: >relevantProjects [](http://example.com/codeflow?start=28&length=16)

NIT: applicableProjects is a bit more expressive on intent.

File: src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs
Comment (aortiz-msft) [inline src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs]: Could we have a private method that simply separates the PackageReference projects from Packages.config projects? Then the logic to warn on one set while use the other set becomes simpler.

**Review by aortiz-msft — APPROVED** [review]
:shipit:

File: src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs
Comment (nkolev92) [reply src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs]: Yep, I can do that. 
Now that I have concept approval, I'll refactor that + try to have 1 method for warning as @erdembayar suggested.

**Comment by nkolev92** [comment]
Welp, the apex test I wrote is failing :D 

@heng-liu Do you know if there's a way to configure the bot to treat a PR that's getting commit updates as not stale?

**Comment by donnie-msft** [comment]
> Do you know if there's a way to configure the bot to treat a PR that's getting commit updates as not stale?

I believe it does consider commits. But you had 8 days pass without a commit, and it's configured to mark 7 day old PRs as stale.

**Comment by nkolev92** [comment]
Oh it actually did.

I see the status label was removed after my force push/commit. 

Thanks for pointing it out @donnie-msft

File: src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs
Comment (donnie-msft) [inline src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs]: Why would this be null/empty? If it is, I'd say still warn anyway and just omit the project names. Otherwise the user may expect something happened, but it did not.

Or... it's impossible for this to be null, so remove the check.

File: src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/Resources.resx
Comment (donnie-msft) [inline src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/Resources.resx]: 0 instead of 1?

File: src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs
Comment (nkolev92) [reply src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs]: Can't be null. I removed the check.

File: src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/Resources.resx
Comment (nkolev92) [reply src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/Resources.resx]: yeah, it should be 0.

File: src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs
Comment (donnie-msft) [inline src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs]: explicit type

**Review by donnie-msft — APPROVED** [review]
Nice! Is there an issue for the docs changes? If so, probably should comment/link it to this issue for reference. If not, please create one.

File: src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs
Comment (donnie-msft) [inline src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs]: explicit type
(sorry, forgot to submit comment here)

**Comment by nkolev92** [comment]
> If so, probably should comment/link it to this issue for reference. If not, please create one.

Yep, I'll create and link the issue. 

My intention is to complete the doc change as soon as possible and possibly before I merge this change, given that it's a sensitive issue.

**Pull request merged**
